File: packages/capacitor/android/src/main/java/com/powersync/capacitor/PowerSyncPlugin.java
package com.powersync.capacitor;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PowerSync")
public class PowerSyncPlugin extends Plugin {

    private PowerSync implementation = new PowerSync();

    @PluginMethod
    public void registerCore(PluginCall call) {
        JSObject ret = new JSObject();
        // For Android we don't statically register the library, so we always return 0
        ret.put("responseCode", 0);
        call.resolve(ret);
    }
}

File: packages/capacitor/ios/Sources/PowerSyncPlugin/PowerSyncPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PowerSyncPlugin)
public class PowerSyncPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "PowerSyncPlugin"
    public let jsName = "PowerSync"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "registerCore", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = PowerSync()

    @objc func registerCore(_ call: CAPPluginCall) {
        call.resolve([
            "responseCode": implementation.registerCore()
        ])
    }
}